File: clients/core/src/managementConsole/applicationAdministration/interfaces/exportedApplicationAnswers.ts
export interface ExportedAnswerColumn {
  questionID: string
  key: string
  title: string
  orderNum: number
  type: 'text' | 'multiselect'
}

export interface ExportedAnswer {
  questionID: string
  answer: string
}

export interface ParticipationExportedAnswers {
  courseParticipationID: string
  answers: ExportedAnswer[]
}

export interface ExportedApplicationAnswersResponse {
  columns: ExportedAnswerColumn[]
  answers: ParticipationExportedAnswers[]
}

File: servers/core/applicationAdministration/applicationDTO/exported_application_answers.go
package applicationDTO

import "github.com/google/uuid"

type ExportedAnswerColumn struct {
	QuestionID uuid.UUID `json:"questionID"`
	Key        string    `json:"key"`
	Title      string    `json:"title"`
	OrderNum   int32     `json:"orderNum"`
	Type       string    `json:"type"`
}

type ExportedAnswer struct {
	QuestionID uuid.UUID `json:"questionID"`
	Answer     string    `json:"answer"`
}

type ParticipationExportedAnswers struct {
	CourseParticipationID uuid.UUID        `json:"courseParticipationID"`
	Answers               []ExportedAnswer `json:"answers"`
}

type ExportedApplicationAnswersResponse struct {
	Columns []ExportedAnswerColumn         `json:"columns"`
	Answers []ParticipationExportedAnswers `json:"answers"`
}
